### PR TITLE
hydra: 2024-08-20 -> 2024-08-21; nixos/hydra: use configured compression in hydra-compress-logs service

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mindavi** [comment]
nxios -> nixos in commit message

If I understand correctly, mtime +3h only compresses logs _older_ than 3h, not _newer_. Source: https://stackoverflow.com/questions/25599094/explaining-the-find-mtime-command#25599430

**Comment by Mindavi** [comment]
Dug up the first commit: https://github.com/NixOS/hydra/commit/b8ff29f0db6d31c70a3c6593277573d4054b46ea

Which already had it. Still not sure about whether we should change it.

**Comment by SuperSandro2000** [comment]
I think I found my miss understanding. I first tried to find files with the zstd extension and thought it wouldn't work but the zst extension is actually used.

File: nixos/modules/services/continuous-integration/hydra/default.nix [hunk 1]
```diff
@@ -500,10 +501,15 @@ in
...
     systemd.services.hydra-compress-logs =
-      { path = [ pkgs.bzip2 ];
+      { path = [ pkgs.bzip2 pkgs.zstd ];
         script =
           ''
-            find /var/lib/hydra/build-logs -type f -name "*.drv" -mtime +3 -size +0c | xargs -r bzip2 -v -f
+            set -eou pipefail
+            compression=$(sed -nr 's/compress_build_logs_compression = ()/\1/p' ${baseDir}/hydra.conf)
+            if [[ $compression == zstd ]]; then
+              compression=zstd --rm
```
Comment (SuperSandro2000) [inline nixos/modules/services/continuous-integration/hydra/default.nix:510]:
```suggestion
              compression="zstd --rm"
```

**Review by Mindavi — APPROVED** [review]
ACK from me to change the name to hydra. Last time it was a bit too quick after the previous rename, but now it's been long enough since: https://github.com/NixOS/nixpkgs/pull/211164

Edit: maybe time to move to by-name?

**Comment by sternenseemann** [comment]
Scripts in `maintainers` have references to hydra, seems like you missed those.

**Comment by Mindavi** [comment]
So that would be the script in `maintainers/scripts/haskell/hydra-report.hs`. Also this started conflicting due to the CVE fixes, but should be easy to resolve, I'd say.

**Review by Mindavi — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
> Scripts in `maintainers` have references to hydra, seems like you missed those.

Ups, those where using an older alias which I did not search for.

**Pull request closed without merging**